editoast/src/converters/generate_routes.rs

```diff
@@ -0,0 +1,423 @@
+                self.add_symetrical_edge(b, u, EdgeType::Track);
+            } else {
+                let u = Node::from_track_endpoint(&buffer.track, Endpoint::End);
+                self.add_symetrical_edge(b, u, EdgeType::Track);
+            }
+        }
+
+        for track in &railjson.track_sections {
+            // We only consider tracks that have no detector on them as we split them
+            if !fwd_detectors.contains_key(&track.id) && !bwd_detectors.contains_key(&track.id) {
+                let u = Node::from_track_endpoint(&track.id, Endpoint::Begin);
+                let v = Node::from_track_endpoint(&track.id, Endpoint::End);
+                self.add_symetrical_edge(u, v, EdgeType::Track);
+            }
+        }
+    }
+
+    fn edges_from_track_sections_links(&mut self, railjson: &RailJson) {
+        for link in &railjson.track_section_links {
+            let u = Node::TrackEndpoint(link.src.clone());
+            let v = Node::TrackEndpoint(link.dst.clone());
+            self.add_symetrical_edge(u, v, EdgeType::SectionLink);
+        }
+    }
+
+    fn edges_from_switches(&mut self, railjson: &RailJson) {
+        for switch in &railjson.switches {
+            let switch_type = railjson
+                .switch_types
+                .iter()
+                .find(|t| t.id == switch.switch_type)
+                .expect("Switch must have associated type");
+
+            for (port_id, switch_ports) in switch_type.groups.iter() {
+                for switch_port in switch_ports {
+                    let u = Node::TrackEndpoint(
+                        switch
+                            .ports
+                            .get(&switch_port.src)
+                            .expect("Switch must have all ports set")
+                            .clone(),
+                    );
+                    let v = Node::TrackEndpoint(
+                        switch
+                            .ports
+                            .get(&switch_port.dst)
+                            .expect("Switch must have all ports set")
+                            .clone(),
+                    );
+                    let edge_type = EdgeType::Switch {
+                        id: switch.id.clone(),
+                        port: port_id.clone(),
+                    };
+                    self.add_symetrical_edge(u, v, edge_type);
+                }
+            }
+        }
+    }
+
+    fn add_directed_edge(&mut self, u: Node, v: Node, edge_type: EdgeType) {
+        self.edges.insert((u.clone(), v.clone()), edge_type);
+        self.successors.entry(u).or_default().push(v);
+    }
+
+    fn add_symetrical_edge(&mut self, u: Node, v: Node, edge_type: EdgeType) {
+        self.add_directed_edge(u.clone(), v.clone(), edge_type.clone());
+        self.add_directed_edge(v, u, edge_type);
+    }
+
+    /* Part 3: compute the different routes */
+
+    // Computes all the routes from one TrackEndpoint to all others
```

Comment:
This function is never called for `TrackEndpoints` specifically, didn't you mean

```suggestion
    // Computes all the routes from one Node to all others
```

Reply:
Indeed!